src/Compilers/VisualBasic/Portable/Binding/Binder_Symbols.vb

```diff
@@ -516,10 +511,15 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 Next
 
                 If forwardedType IsNot Nothing Then
-                    If forwardedType.IsErrorType Then
-                        Dim diagInfo As DiagnosticInfo = DirectCast(forwardedType, ErrorTypeSymbol).ErrorInfo
-                        If diagInfo.Code = ERRID.ERR_TypeFwdCycle2 Then
-                            encounteredCycle = True
+                    If Not reportedAnError And forwardedType.IsErrorType Then
```

Comment:
Why pass two pieces of information about the same thing, allowing people to pass information that can contradict each other. For example, do we handle a case when reportedAnError is false and the bag is null at the same time? Why complicate things when they can be very simple, i.e. simply pass a diagnostics bag when you want to get diagnostics, pass null otherwise.